server/src/test/java/io/crate/integrationtests/GCDanglingArtifactsIntegrationTest.java

```diff
@@ -75,9 +92,15 @@ public void clusterStateProcessed(String source, ClusterState oldState, ClusterS
                     future.onResponse(null);
                 }
             });
-        future.get();
+        future.get(10, TimeUnit.SECONDS);
 
         execute("alter cluster gc dangling artifacts");
-        assertBusy(() -> assertThat(cluster().clusterService().state().metadata().index(danglingIndexName)).isNull());
+        for (String danglingIndexName : danglingIndexNames) {
+            assertBusy(() -> assertThat(cluster().clusterService().state().metadata().index(danglingIndexName)).isNull());
+        }
```

Comment:
very minor: could also move the loop into the assertBusy call as all should be deleted in one state change.